mlir/lib/Dialect/LLVMIR/IR/GENXDialect.cpp

```diff
@@ -60,8 +66,180 @@ LogicalResult GENXDialect::verifyOperationAttribute(Operation *op,
   return success();
 }
 
-#define GET_OP_CLASSES
-#include "mlir/Dialect/LLVMIR/GENXOps.cpp.inc"
+//===----------------------------------------------------------------------===//
+// Type Parsing
+//===----------------------------------------------------------------------===//
 
-#define GET_ATTRDEF_CLASSES
-#include "mlir/Dialect/LLVMIR/GENXOpsAttributes.cpp.inc"
+static Type parseAndVerifyType(GENXDialect const &dialect,
+                               DialectAsmParser &parser) {
+  Type type;
+  SMLoc typeLoc = parser.getCurrentLocation();
+  if (parser.parseType(type))
+    return Type();
+
+  // Allow GENX dialect types
+  if (&type.getDialect() == &dialect)
+    return type;
+
+  // Check other allowed types
+  if (auto t = type.dyn_cast<FloatType>()) {
```

Comment:
These are taken from the SPIRV dialect, Given we generate SPIRV eventually, I think we can have the same type restrictions.

Reply:
May be a comment here (as well as SPIRV dialect) saying that these two sets of restrictions need to be in sync will help in code maintenance.

Thanks

Reply:
It seems to be too restrictive.  PVC mad Op supports C += A*B, where A and B's element types can be int8, uint8, fp16, bf16, tf32, and C's element types can be int32, float, float, float.  Let's allow int8, uint8, fp16, bf16, tf32, int32 and float for matrix.load so we can load A, B and C from memory to register? We can further refine the permissible types for A, B and C in the matrix.mad Op.

Reply:
Sure. The TF32 data type is not yet supported in the LLVM version we are using (it is supported in the most current LLVM version). Will add a TODO for that.

Reply:
done